core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala

```diff
@@ -223,6 +223,7 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Loggi
     logInfo("Connecting to external H2O cluster.")
     val clusterBuildTimeout = conf.cloudTimeout
     val nodes = if (isRestApiBasedClient(hc)) {
+      lockCloud(conf)
```

Comment:
Shouldn't `lockCloud(conf)` be in the below `try` statement in order to handle REST connectivity issues properly?